src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private MyHashMap.Node<K, V>[] table;
+    private int loadFactor;
+    private int capacity;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null || table.length == 0 || size + 1 > loadFactor) {
+            resize();
+        }
+        int indexArrayNode = (key == null) ? 0
+                : (key.hashCode() < 0) ? key.hashCode() % capacity * -1
+                : key.hashCode() % capacity;
+        Node<K, V> newNode = table[indexArrayNode];
+        if (newNode == null) {
+            table[indexArrayNode] = new Node<>(key, value, null);
+        } else {
+            while (true) {
```

Comment:
```suggestion
            while (newNode != null) {
```
I don't think it's good to do an eternal loop, I myself don't know if it's right ;)

Reply:
i too. So let's see if that's possible